staging/src/k8s.io/apiserver/pkg/authorization/union/union.go

```diff
@@ -59,7 +59,7 @@ func (authzHandler unionAuthzHandler) Authorize(ctx context.Context, a authorize
 		}
 		switch decision {
 		case authorizer.DecisionAllow, authorizer.DecisionDeny:
-			return decision, reason, err
+			return decision, reason, utilerrors.NewAggregate(errlist)
```

Comment:
I'm not sure this change is what we want. If an authorizer returns a terminal decision, its error should govern. If all authorizers return NoOpinion, then any errors returned alongside get aggregated. 

It's weird to return a terminal decision from one authorizer and errors from another

Reply:
/hold

Reply:
Sounds reasonable. But looks like this function only take responsible for collecting errors. How about let the caller to decide how to use the error?

Reply:
It's a question of when errors are collected. Aggregating NoOpinion errors to return alongside a NoOpinion decision makes sense. Aggregating NoOpinion errors to return alongside an unrelated Allow or Deny decision does not make sense… in that case the returned errors would have nothing to do with the decision. 

I think the existing behavior is correct. If you want to adjust this PR to document the behavior better in the Godoc, that would be fine. Otherwise, I think we should close this PR

Reply:
Sounds reasonable.